compiler/compile/OMRCompilation.cpp

```diff
@@ -720,7 +720,22 @@ OMR::Compilation::getOSRInductionOffset(TR::Node *node)
 bool
 OMR::Compilation::requiresPreOSRPoint(TR::Node *node)
    {
-   return node->getOpCodeValue() != TR::monent;
+   // When there is no OSR induction offset, a pre OSR point is required
+   // This currently only covers async checks as they save the state when the check is performed
+   if (getOSRInductionOffset(node) == 0)
```

Comment:
The issue here is subtle - asyncchecks are always generated _before_ the side effect of the operation whose bytecode they carry so they have an induction offset of 0. When you have an induction offset of 0 you will always have a prepoint because that is the transition point. For the monent we will transition after the monent, but unlike a call we will not blow the monent into pieces where we could see the pre-osr point on the stack (calls can be inlined). Essentially, this code is saying that if the offset is 0 the prepoint is the point and if the offset is non-zero we need the prepoint if we may need the pre-bytecode execution state as part of setting up a transition which is true for a call, but not for a monent.

All very subtle but I think this is probably the simplest way to encode this notion of do we need two osr points or one.